packages/wallet/src/cip30.ts

```diff
@@ -272,6 +281,11 @@ const baseCip30WalletApi = (
     logger.debug('getting changeAddress');
     try {
       const wallet = await firstValueFrom(wallet$);
+      if (!wallet) {
```

Comment:
@mkazlauskas maybe you could suggest if there is a way not to duplicate same checks in every method? maybe by wrapping it with some proxy handler and checking wallet observable before accessing the exact method?
if it makes sense of course*

Reply:
I suggest a higher order function like `withWallet((wallet) = {...})`